I get some build errors. Those are probably related to my outdated build system ESP-IDF v5.1.5
```
/home/karl/snapclient/components/lightsnapcast/snapcast_protocol_parser.c: In function 'parse_wire_chunk_message':
/home/karl/snapclient/components/lightsnapcast/snapcast_protocol_parser.c:97:7: error: implicit declaration of function 'vTaskDelay' [-Werror=implicit-function-declaration]
   97 |       vTaskDelay(pdMS_TO_TICKS(1));
      |       ^~~~~~~~~~
In file included from /home/karl/snapclient/components/lightsnapcast/snapcast_protocol_parser.c:3:
/home/karl/snapclient/components/lightsnapcast/snapcast_protocol_parser.c: In function 'parse_codec_header_message':
/home/karl/espressif/esp-idf/components/log/include/esp_log.h:265:27: error: format '%s' expects argument of type 'char *', but argument 6 has type 'uint32_t' {aka 'long unsigned int'} [-Werror=format=]
  265 | #define LOG_COLOR(COLOR)  "\033[0;" COLOR "m"
      |                           ^~~~~~~~~
/home/karl/espressif/esp-idf/components/log/include/esp_log.h:268:27: note: in expansion of macro 'LOG_COLOR'
  268 | #define LOG_COLOR_E       LOG_COLOR(LOG_COLOR_RED)
      |                           ^~~~~~~~~
/home/karl/espressif/esp-idf/components/log/include/esp_log.h:282:37: note: in expansion of macro 'LOG_COLOR_E'
  282 | #define LOG_FORMAT(letter, format)  LOG_COLOR_ ## letter #letter " (%" PRIu32 ") %s: " format LOG_RESET_COLOR "\n"
      |                                     ^~~~~~~~~~
/home/karl/espressif/esp-idf/components/log/include/esp_log.h:410:86: note: in expansion of macro 'LOG_FORMAT'
  410 |         if (level==ESP_LOG_ERROR )          { esp_log_write(ESP_LOG_ERROR,      tag, LOG_FORMAT(E, format), esp_log_timestamp(), tag, ##__VA_ARGS__); } \
      |                                                                                      ^~~~~~~~~~
/home/karl/espressif/esp-idf/components/log/include/esp_log.h:432:41: note: in expansion of macro 'ESP_LOG_LEVEL'
  432 |         if ( LOG_LOCAL_LEVEL >= level ) ESP_LOG_LEVEL(level, tag, format, ##__VA_ARGS__); \
      |                                         ^~~~~~~~~~~~~
/home/karl/espressif/esp-idf/components/log/include/esp_log.h:340:38: note: in expansion of macro 'ESP_LOG_LEVEL_LOCAL'
  340 | #define ESP_LOGE( tag, format, ... ) ESP_LOG_LEVEL_LOCAL(ESP_LOG_ERROR,   tag, format, ##__VA_ARGS__)
      |                                      ^~~~~~~~~~~~~~~~~~~
/home/karl/snapclient/components/lightsnapcast/snapcast_protocol_parser.c:200:5: note: in expansion of macro 'ESP_LOGE'
  200 |     ESP_LOGE(TAG, "Codec : %s... not supported %lu", codecStringLen);
      |     ^~~~~~~~
/home/karl/espressif/esp-idf/components/log/include/esp_log.h:265:27: error: format '%lu' expects a matching 'long unsigned int' argument [-Werror=format=]
  265 | #define LOG_COLOR(COLOR)  "\033[0;" COLOR "m"
      |                           ^~~~~~~~~
/home/karl/espressif/esp-idf/components/log/include/esp_log.h:268:27: note: in expansion of macro 'LOG_COLOR'
  268 | #define LOG_COLOR_E       LOG_COLOR(LOG_COLOR_RED)
      |                           ^~~~~~~~~
/home/karl/espressif/esp-idf/components/log/include/esp_log.h:282:37: note: in expansion of macro 'LOG_COLOR_E'
  282 | #define LOG_FORMAT(letter, format)  LOG_COLOR_ ## letter #letter " (%" PRIu32 ") %s: " format LOG_RESET_COLOR "\n"
      |                                     ^~~~~~~~~~
/home/karl/espressif/esp-idf/components/log/include/esp_log.h:410:86: note: in expansion of macro 'LOG_FORMAT'
  410 |         if (level==ESP_LOG_ERROR )          { esp_log_write(ESP_LOG_ERROR,      tag, LOG_FORMAT(E, format), esp_log_timestamp(), tag, ##__VA_ARGS__); } \
      |                                                                                      ^~~~~~~~~~
/home/karl/espressif/esp-idf/components/log/include/esp_log.h:432:41: note: in expansion of macro 'ESP_LOG_LEVEL'
  432 |         if ( LOG_LOCAL_LEVEL >= level ) ESP_LOG_LEVEL(level, tag, format, ##__VA_ARGS__); \
      |                                         ^~~~~~~~~~~~~
/home/karl/espressif/esp-idf/components/log/include/esp_log.h:340:38: note: in expansion of macro 'ESP_LOG_LEVEL_LOCAL'
  340 | #define ESP_LOGE( tag, format, ... ) ESP_LOG_LEVEL_LOCAL(ESP_LOG_ERROR,   tag, format, ##__VA_ARGS__)
      |                                      ^~~~~~~~~~~~~~~~~~~
/home/karl/snapclient/components/lightsnapcast/snapcast_protocol_parser.c:200:5: note: in expansion of macro 'ESP_LOGE'
  200 |     ESP_LOGE(TAG, "Codec : %s... not supported %lu", codecStringLen);
      |     ^~~~~~~~
/home/karl/espressif/esp-idf/components/log/include/esp_log.h:265:27: error: format '%s' expects argument of type 'char *', but argument 6 has type 'uint32_t' {aka 'long unsigned int'} [-Werror=format=]
  265 | #define LOG_COLOR(COLOR)  "\033[0;" COLOR "m"
      |                           ^~~~~~~~~
/home/karl/espressif/esp-idf/components/log/include/esp_log.h:269:27: note: in expansion of macro 'LOG_COLOR'
  269 | #define LOG_COLOR_W       LOG_COLOR(LOG_COLOR_BROWN)
      |                           ^~~~~~~~~
/home/karl/espressif/esp-idf/components/log/include/esp_log.h:282:37: note: in expansion of macro 'LOG_COLOR_W'
  282 | #define LOG_FORMAT(letter, format)  LOG_COLOR_ ## letter #letter " (%" PRIu32 ") %s: " format LOG_RESET_COLOR "\n"
      |                                     ^~~~~~~~~~
/home/karl/espressif/esp-idf/components/log/include/esp_log.h:411:86: note: in expansion of macro 'LOG_FORMAT'
  411 |         else if (level==ESP_LOG_WARN )      { esp_log_write(ESP_LOG_WARN,       tag, LOG_FORMAT(W, format), esp_log_timestamp(), tag, ##__VA_ARGS__); } \
      |                                                                                      ^~~~~~~~~~
/home/karl/espressif/esp-idf/components/log/include/esp_log.h:432:41: note: in expansion of macro 'ESP_LOG_LEVEL'
  432 |         if ( LOG_LOCAL_LEVEL >= level ) ESP_LOG_LEVEL(level, tag, format, ##__VA_ARGS__); \
      |                                         ^~~~~~~~~~~~~
/home/karl/espressif/esp-idf/components/log/include/esp_log.h:340:38: note: in expansion of macro 'ESP_LOG_LEVEL_LOCAL'
  340 | #define ESP_LOGE( tag, format, ... ) ESP_LOG_LEVEL_LOCAL(ESP_LOG_ERROR,   tag, format, ##__VA_ARGS__)
      |                                      ^~~~~~~~~~~~~~~~~~~
/home/karl/snapclient/components/lightsnapcast/snapcast_protocol_parser.c:200:5: note: in expansion of macro 'ESP_LOGE'
  200 |     ESP_LOGE(TAG, "Codec : %s... not supported %lu", codecStringLen);
      |     ^~~~~~~~
/home/karl/espressif/esp-idf/components/log/include/esp_log.h:265:27: error: format '%lu' expects a matching 'long unsigned int' argument [-Werror=format=]
  265 | #define LOG_COLOR(COLOR)  "\033[0;" COLOR "m"
      |                           ^~~~~~~~~
/home/karl/espressif/esp-idf/components/log/include/esp_log.h:269:27: note: in expansion of macro 'LOG_COLOR'
  269 | #define LOG_COLOR_W       LOG_COLOR(LOG_COLOR_BROWN)
      |                           ^~~~~~~~~
/home/karl/espressif/esp-idf/components/log/include/esp_log.h:282:37: note: in expansion of macro 'LOG_COLOR_W'
  282 | #define LOG_FORMAT(letter, format)  LOG_COLOR_ ## letter #letter " (%" PRIu32 ") %s: " format LOG_RESET_COLOR "\n"
      |                                     ^~~~~~~~~~
/home/karl/espressif/esp-idf/components/log/include/esp_log.h:411:86: note: in expansion of macro 'LOG_FORMAT'
  411 |         else if (level==ESP_LOG_WARN )      { esp_log_write(ESP_LOG_WARN,       tag, LOG_FORMAT(W, format), esp_log_timestamp(), tag, ##__VA_ARGS__); } \
      |                                                                                      ^~~~~~~~~~
/home/karl/espressif/esp-idf/components/log/include/esp_log.h:432:41: note: in expansion of macro 'ESP_LOG_LEVEL'
  432 |         if ( LOG_LOCAL_LEVEL >= level ) ESP_LOG_LEVEL(level, tag, format, ##__VA_ARGS__); \
      |                                         ^~~~~~~~~~~~~
/home/karl/espressif/esp-idf/components/log/include/esp_log.h:340:38: note: in expansion of macro 'ESP_LOG_LEVEL_LOCAL'
  340 | #define ESP_LOGE( tag, format, ... ) ESP_LOG_LEVEL_LOCAL(ESP_LOG_ERROR,   tag, format, ##__VA_ARGS__)
      |                                      ^~~~~~~~~~~~~~~~~~~
/home/karl/snapclient/components/lightsnapcast/snapcast_protocol_parser.c:200:5: note: in expansion of macro 'ESP_LOGE'
  200 |     ESP_LOGE(TAG, "Codec : %s... not supported %lu", codecStringLen);
      |     ^~~~~~~~
/home/karl/snapclient/components/lightsnapcast/snapcast_protocol_parser.c:200:1: error: format '%s' expects argument of type 'char *', but argument 6 has type 'uint32_t' {aka 'long unsigned int'} [-Werror=format=]
  200 |     ESP_LOGE(TAG, "Codec : %s... not supported %lu", codecStringLen);
      | ^   ~~~~~~~
/home/karl/espressif/esp-idf/components/log/include/esp_log.h:282:59: note: in definition of macro 'LOG_FORMAT'
  282 | #define LOG_FORMAT(letter, format)  LOG_COLOR_ ## letter #letter " (%" PRIu32 ") %s: " format LOG_RESET_COLOR "\n"
      |                                                           ^~~~~~
/home/karl/espressif/esp-idf/components/log/include/esp_log.h:432:41: note: in expansion of macro 'ESP_LOG_LEVEL'
  432 |         if ( LOG_LOCAL_LEVEL >= level ) ESP_LOG_LEVEL(level, tag, format, ##__VA_ARGS__); \
      |                                         ^~~~~~~~~~~~~
/home/karl/espressif/esp-idf/components/log/include/esp_log.h:340:38: note: in expansion of macro 'ESP_LOG_LEVEL_LOCAL'
  340 | #define ESP_LOGE( tag, format, ... ) ESP_LOG_LEVEL_LOCAL(ESP_LOG_ERROR,   tag, format, ##__VA_ARGS__)
      |                                      ^~~~~~~~~~~~~~~~~~~
/home/karl/snapclient/components/lightsnapcast/snapcast_protocol_parser.c:200:5: note: in expansion of macro 'ESP_LOGE'
  200 |     ESP_LOGE(TAG, "Codec : %s... not supported %lu", codecStringLen);
      |     ^~~~~~~~
/home/karl/snapclient/components/lightsnapcast/snapcast_protocol_parser.c:200:1: error: format '%lu' expects a matching 'long unsigned int' argument [-Werror=format=]
  200 |     ESP_LOGE(TAG, "Codec : %s... not supported %lu", codecStringLen);
      | ^   ~~~~~~~
/home/karl/espressif/esp-idf/components/log/include/esp_log.h:282:59: note: in definition of macro 'LOG_FORMAT'
  282 | #define LOG_FORMAT(letter, format)  LOG_COLOR_ ## letter #letter " (%" PRIu32 ") %s: " format LOG_RESET_COLOR "\n"
      |                                                           ^~~~~~
/home/karl/espressif/esp-idf/components/log/include/esp_log.h:432:41: note: in expansion of macro 'ESP_LOG_LEVEL'
  432 |         if ( LOG_LOCAL_LEVEL >= level ) ESP_LOG_LEVEL(level, tag, format, ##__VA_ARGS__); \
      |                                         ^~~~~~~~~~~~~
/home/karl/espressif/esp-idf/components/log/include/esp_log.h:340:38: note: in expansion of macro 'ESP_LOG_LEVEL_LOCAL'
  340 | #define ESP_LOGE( tag, format, ... ) ESP_LOG_LEVEL_LOCAL(ESP_LOG_ERROR,   tag, format, ##__VA_ARGS__)
      |                                      ^~~~~~~~~~~~~~~~~~~
/home/karl/snapclient/components/lightsnapcast/snapcast_protocol_parser.c:200:5: note: in expansion of macro 'ESP_LOGE'
  200 |     ESP_LOGE(TAG, "Codec : %s... not supported %lu", codecStringLen);
      |     ^~~~~~~~
/home/karl/snapclient/components/lightsnapcast/snapcast_protocol_parser.c:200:1: error: format '%s' expects argument of type 'char *', but argument 6 has type 'uint32_t' {aka 'long unsigned int'} [-Werror=format=]
  200 |     ESP_LOGE(TAG, "Codec : %s... not supported %lu", codecStringLen);
      | ^   ~~~~~~~
/home/karl/espressif/esp-idf/components/log/include/esp_log.h:282:59: note: in definition of macro 'LOG_FORMAT'
  282 | #define LOG_FORMAT(letter, format)  LOG_COLOR_ ## letter #letter " (%" PRIu32 ") %s: " format LOG_RESET_COLOR "\n"
      |                                                           ^~~~~~
/home/karl/espressif/esp-idf/components/log/include/esp_log.h:432:41: note: in expansion of macro 'ESP_LOG_LEVEL'
  432 |         if ( LOG_LOCAL_LEVEL >= level ) ESP_LOG_LEVEL(level, tag, format, ##__VA_ARGS__); \
      |                                         ^~~~~~~~~~~~~
/home/karl/espressif/esp-idf/components/log/include/esp_log.h:340:38: note: in expansion of macro 'ESP_LOG_LEVEL_LOCAL'
  340 | #define ESP_LOGE( tag, format, ... ) ESP_LOG_LEVEL_LOCAL(ESP_LOG_ERROR,   tag, format, ##__VA_ARGS__)
      |                                      ^~~~~~~~~~~~~~~~~~~
/home/karl/snapclient/components/lightsnapcast/snapcast_protocol_parser.c:200:5: note: in expansion of macro 'ESP_LOGE'
  200 |     ESP_LOGE(TAG, "Codec : %s... not supported %lu", codecStringLen);
      |     ^~~~~~~~
/home/karl/snapclient/components/lightsnapcast/snapcast_protocol_parser.c:200:1: error: format '%lu' expects a matching 'long unsigned int' argument [-Werror=format=]
  200 |     ESP_LOGE(TAG, "Codec : %s... not supported %lu", codecStringLen);
      | ^   ~~~~~~~
/home/karl/espressif/esp-idf/components/log/include/esp_log.h:282:59: note: in definition of macro 'LOG_FORMAT'
  282 | #define LOG_FORMAT(letter, format)  LOG_COLOR_ ## letter #letter " (%" PRIu32 ") %s: " format LOG_RESET_COLOR "\n"
      |                                                           ^~~~~~
/home/karl/espressif/esp-idf/components/log/include/esp_log.h:432:41: note: in expansion of macro 'ESP_LOG_LEVEL'
  432 |         if ( LOG_LOCAL_LEVEL >= level ) ESP_LOG_LEVEL(level, tag, format, ##__VA_ARGS__); \
      |                                         ^~~~~~~~~~~~~
/home/karl/espressif/esp-idf/components/log/include/esp_log.h:340:38: note: in expansion of macro 'ESP_LOG_LEVEL_LOCAL'
  340 | #define ESP_LOGE( tag, format, ... ) ESP_LOG_LEVEL_LOCAL(ESP_LOG_ERROR,   tag, format, ##__VA_ARGS__)
      |                                      ^~~~~~~~~~~~~~~~~~~
/home/karl/snapclient/components/lightsnapcast/snapcast_protocol_parser.c:200:5: note: in expansion of macro 'ESP_LOGE'
  200 |     ESP_LOGE(TAG, "Codec : %s... not supported %lu", codecStringLen);
      |     ^~~~~~~~
/home/karl/espressif/esp-idf/components/log/include/esp_log.h:265:27: error: format '%s' expects argument of type 'char *', but argument 6 has type 'uint32_t' {aka 'long unsigned int'} [-Werror=format=]
  265 | #define LOG_COLOR(COLOR)  "\033[0;" COLOR "m"
      |                           ^~~~~~~~~
/home/karl/espressif/esp-idf/components/log/include/esp_log.h:270:27: note: in expansion of macro 'LOG_COLOR'
  270 | #define LOG_COLOR_I       LOG_COLOR(LOG_COLOR_GREEN)
      |                           ^~~~~~~~~
/home/karl/espressif/esp-idf/components/log/include/esp_log.h:282:37: note: in expansion of macro 'LOG_COLOR_I'
  282 | #define LOG_FORMAT(letter, format)  LOG_COLOR_ ## letter #letter " (%" PRIu32 ") %s: " format LOG_RESET_COLOR "\n"
      |                                     ^~~~~~~~~~
/home/karl/espressif/esp-idf/components/log/include/esp_log.h:414:86: note: in expansion of macro 'LOG_FORMAT'
  414 |         else                                { esp_log_write(ESP_LOG_INFO,       tag, LOG_FORMAT(I, format), esp_log_timestamp(), tag, ##__VA_ARGS__); } \
      |                                                                                      ^~~~~~~~~~
/home/karl/espressif/esp-idf/components/log/include/esp_log.h:432:41: note: in expansion of macro 'ESP_LOG_LEVEL'
  432 |         if ( LOG_LOCAL_LEVEL >= level ) ESP_LOG_LEVEL(level, tag, format, ##__VA_ARGS__); \
      |                                         ^~~~~~~~~~~~~
/home/karl/espressif/esp-idf/components/log/include/esp_log.h:340:38: note: in expansion of macro 'ESP_LOG_LEVEL_LOCAL'
  340 | #define ESP_LOGE( tag, format, ... ) ESP_LOG_LEVEL_LOCAL(ESP_LOG_ERROR,   tag, format, ##__VA_ARGS__)
      |                                      ^~~~~~~~~~~~~~~~~~~
/home/karl/snapclient/components/lightsnapcast/snapcast_protocol_parser.c:200:5: note: in expansion of macro 'ESP_LOGE'
  200 |     ESP_LOGE(TAG, "Codec : %s... not supported %lu", codecStringLen);
      |     ^~~~~~~~
/home/karl/espressif/esp-idf/components/log/include/esp_log.h:265:27: error: format '%lu' expects a matching 'long unsigned int' argument [-Werror=format=]
  265 | #define LOG_COLOR(COLOR)  "\033[0;" COLOR "m"
      |                           ^~~~~~~~~
/home/karl/espressif/esp-idf/components/log/include/esp_log.h:270:27: note: in expansion of macro 'LOG_COLOR'
  270 | #define LOG_COLOR_I       LOG_COLOR(LOG_COLOR_GREEN)
      |                           ^~~~~~~~~
/home/karl/espressif/esp-idf/components/log/include/esp_log.h:282:37: note: in expansion of macro 'LOG_COLOR_I'
  282 | #define LOG_FORMAT(letter, format)  LOG_COLOR_ ## letter #letter " (%" PRIu32 ") %s: " format LOG_RESET_COLOR "\n"
      |                                     ^~~~~~~~~~
/home/karl/espressif/esp-idf/components/log/include/esp_log.h:414:86: note: in expansion of macro 'LOG_FORMAT'
  414 |         else                                { esp_log_write(ESP_LOG_INFO,       tag, LOG_FORMAT(I, format), esp_log_timestamp(), tag, ##__VA_ARGS__); } \
      |                                                                                      ^~~~~~~~~~
/home/karl/espressif/esp-idf/components/log/include/esp_log.h:432:41: note: in expansion of macro 'ESP_LOG_LEVEL'
  432 |         if ( LOG_LOCAL_LEVEL >= level ) ESP_LOG_LEVEL(level, tag, format, ##__VA_ARGS__); \
      |                                         ^~~~~~~~~~~~~
/home/karl/espressif/esp-idf/components/log/include/esp_log.h:340:38: note: in expansion of macro 'ESP_LOG_LEVEL_LOCAL'
  340 | #define ESP_LOGE( tag, format, ... ) ESP_LOG_LEVEL_LOCAL(ESP_LOG_ERROR,   tag, format, ##__VA_ARGS__)
      |                                      ^~~~~~~~~~~~~~~~~~~
/home/karl/snapclient/components/lightsnapcast/snapcast_protocol_parser.c:200:5: note: in expansion of macro 'ESP_LOGE'
  200 |     ESP_LOGE(TAG, "Codec : %s... not supported %lu", codecStringLen);
      |     ^~~~~~~~
/home/karl/snapclient/components/lightsnapcast/snapcast_protocol_parser.c:216:7: error: implicit declaration of function 'strcmp' [-Werror=implicit-function-declaration]
  216 |   if (strcmp(codecString, "opus") == 0) {
      |       ^~~~~~
/home/karl/snapclient/components/lightsnapcast/snapcast_protocol_parser.c:4:1: note: include '<string.h>' or provide a declaration of 'strcmp'
    3 | #include "esp_log.h"
  +++ |+#include <string.h>

```

Using IDF v5.5.3 it builds without issues. Could yo still resolve those errors please?

--- a/components/lightsnapcast/snapcast_protocol_parser.c
+++ b/components/lightsnapcast/snapcast_protocol_parser.c
@@ -1,6 +1,8 @@
 #include "snapcast_protocol_parser.h"
 
 #include "esp_log.h"
+#include "freertos/task.h" // for vTaskDelay
+#include <string.h>
 
 // MACROS for reading from connection
 #define READ_BYTE(parser, dest) \
@@ -197,7 +199,7 @@ parser_return_state_t parse_codec_header_message(
   if (codecStringLen + 1 > sizeof(codecString)) {
     READ_DATA(parser, codecString, sizeof(codecString)-1);
     codecString[sizeof(codecString)-1] = 0; // null terminate
-    ESP_LOGE(TAG, "Codec : %s... not supported %lu", codecStringLen);
+    ESP_LOGE(TAG, "Codec : %s... not supported", codecString);
     ESP_LOGI(TAG,
              "Change encoder codec to "
              "opus, flac or pcm in "

--- a/main/connection_handler.c
+++ b/main/connection_handler.c
@@ -328,7 +328,7 @@ static int fill_buffer(bool* first_netbuf_processed, int* rc1,
 
     *rc1 = netbuf_data(firstNetBuf, (void**)start, len);
     if (*rc1 == ERR_OK) {
-      ESP_LOGD(TAG, "netconn rx, data len: %d, %d", len,
+      ESP_LOGD(TAG, "netconn rx, data len: %d, %d", *len,
                netbuf_len(firstNetBuf));
       return 0;
     } else {

